### ISSUE TITLE
nuitka compiled exe in Windows is indentified as a virus/trojan

### ISSUE THREAD

**Issue by Raviu56** [Nuitka/Nuitka#2701]
**Any Onefile compiled python code in Windows is identified as a virus/trojan by avast and many others.**

_Some standalone builds are also reported as viruses/Trojans but not all seems depending on specific code compiled_

I know that there are already reported similar issues but they're old, and some does not give code examples

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version
```
1.9.7
Commercial: None
Python: 3.8.18 (default, Aug 28 2023, 15:16:02) [MSC v.1900 64 bit (AMD64)]
Flavor: CPython Official
Executable: C:\Program Files\Python38\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: ~\gcc (gcc 12.1.0).
```
Actually it's Windows 11 Pro

- How did you install Nuitka and Python
Using installer from here 
https://github.com/adang1345/PythonWindows/blob/master/3.8.18/python-3.8.18-amd64-full.exe
Here are it virustotal report
https://www.virustotal.com/gui/file/c61ab0ecc6a529492015d29c3c9e15e5d4340e78c7e232fe59b0d5d870eb02e2
- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip freeze
```
gmsh==4.12.0
Nuitka==1.9.7
numpy==1.24.4
numpy-stl==3.1.1
ordered-set==4.1.0
parea==0.1.1
psutil==5.9.7
PyQt5==5.15.10
PyQt5-Qt5==5.15.2
PyQt5-sip==12.13.0
python-utils==3.8.1
screeninfo==0.8.1
shapely==2.0.2
trimesh==4.0.8
typing_extensions==4.9.0
zstandard==0.22.0
```

- Many times when you get an error from Nuitka, your setup may be special
Any python code compiled as OneFile in Windows is reported  as virus/trojan. If the same file is build as standalone no problem is reported when running with anti-virus installed.

- Also supply a Short, Self Contained, Correct, Example
Call this code `mypython_version.py` and build it as a very simple example:
```
import sys

print("Python version:")
print(sys.version)
```
- Provide in your issue the Nuitka options used
`python -m nuitka --onefile --remove-output mypython_version.py`

Here is its virustotal report
https://www.virustotal.com/gui/file/bbd9f9e104cf6bc334aa57dcf002b5ca619fa3c97f8ff540a9734d0e4a1e71d5

Compiling same as a standalone, include folder in a zip file upload to virustotal shows a similar problem too:
https://www.virustotal.com/gui/file/bdf60fcdca73e9f4deb58ec6ec720560ea772bf5da185a0e893a592d44e74779

Uploading a single exe from inside standalone build:
https://www.virustotal.com/gui/file/f2dc4a670a46ba80a6ad9a5f61a07c543dd8677dfa0bc2da660bb37fd3be6597

This is a another python example that's not reported a virus if built standalone for major anti-virus program like avast despite having more complex code and dependencies, report is much cleaner but not 100% on virustotal. But it's reported so if built as onefile:
Call it p-area.py and compile as `python -m nuitka --standalone --remove-output --output-filename=parea p-area.py` or `python -m nuitka --onefile --remove-output --output-filename=parea p-area.py`
```
# -*- coding: utf-8 -*-
import re
import sys
from parea import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```
Here is its standalone report
https://www.virustotal.com/gui/file/5e032f573c07c625304dcc1b076e49c79a09a7497bf6e19aa6e93e4585559eee

Here is its onefile report
https://www.virustotal.com/gui/file/3c7b754e63968d032332b87f158cf378021470484941ae5d22759eb2fff72991

**Comment by 7gxycn08** [comment]
`pip uninstall zstandard`

Then manually pack with upx.

**Comment by kayhayen** [comment]
Nuitka is the preferred choice of malware authors due to its protection of the source code, and the inability to investigate behavior of the program without deep analysis. This leads to Nuitka compiled binaries being reported as malware based on being similar to it, even for small programs, esp. for programs that use Crypto libraries and that deal with Crypto payments in particular.

There are ways around this, but further action will be needed, they are described here: https://nuitka.net/doc/user-manual.html#windows-virus-scanners

Naive compilation however is exactly what malware authors do, so it's not going to be sufficient to get your binaries not flagged.

**Comment by MyraBaba** [comment]
> `pip uninstall zstandard`
> 
> 1. compile manually with nuitka.
> 2. Then manually pack with upx.

@7gxycn08 

hi  how to compile ? with onefile o standalone? if standaalone how pack to one file all dll and exe ?
